[BUG] `tests/cli_smoke_test.py::test_update_check_smoke_uses_local_stub` fails in restricted environments when the fixture tries to bind a localhost HTTP server

### Summary

The smoke test `tests/cli_smoke_test.py::test_update_check_smoke_uses_local_stub` fails during fixture setup, before the CLI update-check behavior is exercised. The `release_api_url` fixture creates a real `ThreadingHTTPServer` bound to `127.0.0.1`, and in restricted environments that socket bind is denied with `PermissionError: [Errno 1] Operation not permitted`.

### Expected vs actual behavior

Expected:
- The smoke test should validate the CLI update-check flow using a local stub server.
- If the environment does not allow binding a localhost socket, the test should skip gracefully instead of failing the suite.

Actual:
- The fixture fails while creating `ThreadingHTTPServer(("127.0.0.1", 0), _ReleaseHandler)`.
- The test errors during setup.
- `make test-cov` fails even though the underlying CLI logic is not what broke.

### Steps to reproduce

1. Use the project virtual environment.
2. Run:
   ```bash
   python -m pytest tests/cli_smoke_test.py::test_update_check_smoke_uses_local_stub -q
3. Observe the failure during fixture setup in release_api_url at tests/cli_smoke_test.py:334

### Can you reproduce it consistently?

Yes

### How often does it occur?

Every time

### Operating system

macOS

### Logs and error output

```
  tests/cli_smoke_test.py:334: in release_api_url
      server = ThreadingHTTPServer(("127.0.0.1", 0), _ReleaseHandler)
  ...
  self.socket.bind(self.server_address)
  E   PermissionError: [Errno 1] Operation not permitted
```


### Additional context

This was discovered while running the project quality checks from Codex. The full test suite was executed via:

```source venv/bin/activate && make lint && make format-check && make typecheck && make test-cov```

All earlier checks passed, but make test-cov failed because this smoke test tried to start a localhost HTTP server in its fixture. To confirm it was isolated to this test, I reran:

```source venv/bin/activate && python -m pytest tests/cli_smoke_test.py::test_update_check_smoke_uses_local_stub -q```

That reproduced the same error immediately.

This is an environment-sensitive test bug rather than a product bug. The likely fix is to update the release_api_url fixture in tests/cli_smoke_test.py (line 334) to catch PermissionError and skip the test when localhost socket binding is not permitted.

My plan to solve:

Update the fixture in `tests/cli_smoke_test.py` at line 332 by wrapping the `ThreadingHTTPServer(("127.0.0.1", 0), _ReleaseHandler)` initialization in a `try/except PermissionError` block; if binding fails, call `pytest.skip(...)` with a clear, descriptive message so the test is skipped rather than failing. After making this change, verify the targeted test by activating the virtual environment with `source venv/bin/activate` and running `python -m pytest tests/cli_smoke_test.py::test_update_check_smoke_uses_local_stub -q`, where the expected result in this environment should be a skipped test instead of an error. Finally, re-run the full suite gate using `make lint && make format-check && make typecheck && make test-cov`, confirming that this test no longer causes `make test-cov` to fail.

--- a/tests/cli_smoke_test.py
+++ b/tests/cli_smoke_test.py
@@ -342,7 +342,13 @@ class _ReleaseHandler(BaseHTTPRequestHandler):
 
 @pytest.fixture()
 def release_api_url() -> str:
-    server = ThreadingHTTPServer(("127.0.0.1", 0), _ReleaseHandler)
+    try:
+        server = ThreadingHTTPServer(("127.0.0.1", 0), _ReleaseHandler)
+    except OSError as exc:
+        if exc.errno in {errno.EPERM, errno.EACCES}:
+            pytest.skip("localhost HTTP server binding is not permitted in this environment")
+        raise
+
     thread = Thread(target=server.serve_forever, daemon=True)
     thread.start()
     try:

